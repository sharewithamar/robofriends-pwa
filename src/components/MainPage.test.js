import { shallow } from 'enzyme';
import React from 'react';
import MainPage from './MainPage';

let wrapper;
beforeEach(() => {
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [],
    searchField: '',
    isPending: false
  };
  wrapper = shallow(<MainPage {...mockProps} />);
});

it('renders Mainpage without crashing', () => {
  expect(wrapper.debug()).toMatchSnapshot();
});

it('filters robots correctly', () => {
  expect(wrapper.instance().filterRobots()).toEqual([]);
});

it('filters robots correctly 2', () => {
  const mockProps2 = {
    onRequestRobots: jest.fn(),
    robots: [{ id: 1, name: 'John', email: 'john@example.com' }],
    searchField: 'J',
    isPending: false
  };
  const wrapper2 = shallow(<MainPage {...mockProps2} />);
  expect(wrapper2.instance().filterRobots()).toEqual([
    { id: 1, name: 'John', email: 'john@example.com' }
  ]);
});

it('filters robots correctly 3', () => {
  const mockProps2 = {
    onRequestRobots: jest.fn(),
    robots: [{ id: 1, name: 'John', email: 'john@example.com' }],
    searchField: 'a',
    isPending: false
  };
  const wrapper2 = shallow(<MainPage {...mockProps2} />);
  expect(wrapper2.instance().filterRobots()).toEqual([]);
});
