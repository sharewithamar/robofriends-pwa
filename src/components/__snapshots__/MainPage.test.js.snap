// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Mainpage without crashing 1`] = `
"<div className=\\"tc\\">
  <Header />
  <SearchBox searchChange={[undefined]} />
  <Scroll>
    <ErrorBoundary>
      <CardList robots={[Function]} />
    </ErrorBoundary>
  </Scroll>
</div>"
`;
